src/models/modelsRestApiClient.ts

```diff
@@ -0,0 +1,370 @@
+    const deleteQuery = `DROP MODEL ${mysql.escapeId(project)}.${mysql.escapeId(
+      name
+    )}`;
+    await this.sqlClient.runQuery(deleteQuery);
+  }
+
+  /**
+   * Queries this model for a single prediction. For batch predictions, use batchQuery.
+   * @param {string} name - Name of the model.
+   * @param {string} targetColumn - Column that the model predicts.
+   * @param {string} project - Project the model belongs to.
+   * @param {QueryOptions} options - Options to use when querying the model.
+   * @returns {Promise<ModelPrediction | undefined>} - The prediction result, or undefined if the model isn't valid on the backend.
+   */
+  override async queryModel(
+    name: string,
+    targetColumn: string,
+    project: string,
+    options: QueryOptions
+  ): Promise<ModelPrediction | undefined> {
+    const selectClause = `SELECT * FROM ${mysql.escapeId(
+      project
+    )}.${mysql.escapeId(name)}`;
+    const whereClause = this.makeWhereClause(options['where'] || []);
+    const selectQuery = [selectClause, whereClause].join('\n');
+    const sqlQueryResult = await this.sqlClient.runQuery(selectQuery);
+    if (sqlQueryResult.rows.length === 0) {
+      return undefined;
+    }
+    const predictionRow = sqlQueryResult.rows[0];
+    const prediction = {
+      value: predictionRow[targetColumn],
+      explain: JSON.parse(predictionRow[`${targetColumn}_explain`] || '{}'),
+      data: predictionRow,
+    };
+    return prediction;
+  }
+
+  /**
+   * Queries this model for a batch prediction by joining with another data source.
+   * @param {string} name - Name of the model.
+   * @param {string} targetColumn - Column that the model predicts.
+   * @param {string} project - Project the model belongs to.
+   * @param {BatchQueryOptions} options - Options to use when querying the model.
+   * @returns {Promise<Array<ModelPrediction>>} - All prediction results from the batch query.
+   */
+  override async batchQueryModel(
+    name: string,
+    targetColumn: string,
+    project: string,
+    options: BatchQueryOptions
+  ): Promise<Array<ModelPrediction>> {
+    const selectClause = `SELECT m.${mysql.escapeId(
+      targetColumn
+    )} AS predicted, m.*, t.*`;
+    const fromClause = `FROM ${mysql.escapeId(project)}.${mysql.escapeId(
+      name
+    )} AS m`;
+    const joinId = options['join'];
+    const joinClause = `JOIN ${mysql.escapeId(joinId)} AS t`;
+    const whereClause = this.makeWhereClause(options['where'] || []);
+    const limitClause = options['limit']
+      ? `LIMIT ${mysql.escape(options['limit'])}`
+      : '';
+    const selectQuery = [
+      selectClause,
+      fromClause,
+      joinClause,
+      whereClause,
+      limitClause,
+    ].join('\n');
+    const sqlQueryResult = await this.sqlClient.runQuery(selectQuery);
+    if (sqlQueryResult.rows.length === 0) {
+      return [];
+    }
+    return sqlQueryResult.rows.map((r) => {
```

Comment:
Avoiding a redundant reading take advantage on te arraow functions executing without return

```suggestion
    return sqlQueryResult.rows.map((r) => (
      {
        value: r['predicted'],
        explain: JSON.parse(r[`${targetColumn}_explain`] || '{}'),
        data: r,
      }
    ));
```